Description: Link libaudioroute with Expat
 The library uses Expat APIs to parse mixer configuration files.  Resolve
 those symbols at link time so the shared library records its runtime
 dependency on libexpat.
Author: Lukas Märdian <lukas.maerdian@oss.qualcomm.com>
Forwarded: https://github.com/AudioReach/audioreach-audio-utils/pull/49
Last-Update: 2026-09-21

--- a/audio-route/Makefile.am
+++ b/audio-route/Makefile.am
@@ -28,8 +28,8 @@ system_include_HEADERS = $(system_sources)
 
 lib_LTLIBRARIES = libaudioroute.la
 libaudioroute_la_SOURCES := $(c_sources) $(system_source)
-libaudioroute_la_CFLAGS := $(AM_CFLAGS)
+libaudioroute_la_CFLAGS := $(AM_CFLAGS) $(EXPAT_CFLAGS)
 libaudioroute_la_CFLAGS += -D__unused=__attribute__\(\(__unused__\)\)
 libaudioroute_la_CFLAGS += -DNDEBUG
-libaudioroute_la_LIBADD := -ltinyalsa
+libaudioroute_la_LIBADD := -ltinyalsa $(EXPAT_LIBS)
 libaudioroute_la_LDFLAGS := -shared -version-number @LT_VERSION_NUMBER@
--- a/audio-route/configure.ac
+++ b/audio-route/configure.ac
@@ -19,6 +19,8 @@ AC_PROG_MAKE_SET
 AC_PROG_MAKE_SET
 PKG_PROG_PKG_CONFIG
 
+PKG_CHECK_MODULES([EXPAT], [expat])
+
 m4_define([LT_MAJOR], [1])
 m4_define([LT_MINOR], [0])
 m4_define([LT_PATCH], [0])
